classes/ActionScheduler_PastDueMonitor.php

```diff
@@ -0,0 +1,363 @@
+	 */
+	protected $threshold_minimum;
+
+	/**
+	 * Number of minimum past-due actions to send email notice.
+	 *
+	 * @var int
+	 */
+	protected $threshold_email_minimum;
+
+	/**
+	 * Number of seconds before past-due actions check after
+	 * negative (not flooded) check.
+	 *
+	 * @var int
+	 */
+	protected $interval_check;
+
+	/**
+	 * Number of seconds between email notices.
+	 *
+	 * @var int
+	 */
+	protected $interval_email_seconds;
+
+	/**
+	 * Number of past-due actions (determined by thresholds).
+	 *
+	 * @var int
+	 */
+	protected $num_pastdue_actions = 0;
+
+	/**
+	 * Query arguments for past-due actions.
+	 *
+	 * @var array<string, string|int>
+	 */
+	protected $query_args = array();
+
+	/**
+	 * Notification methods.
+	 *
+	 * @var array<string, bool>
+	 */
+	protected $notify_methods = array();
+
+	/**
+	 * Email address to send notification.
+	 *
+	 * @var string
+	 */
+	protected $notify_email_to;
+
+	/**
+	 * Get singleton instance.
+	 *
+	 * @return ActionScheduler_PastDueMonitor
+	 *
+	 * @codeCoverageIgnore
+	 */
+	public static function instance() {
+
+		if ( empty( self::$monitor ) ) {
+			$class         = apply_filters( 'action_scheduler_pastdue_actions_monitor_class', 'ActionScheduler_PastDueMonitor' );
+			self::$monitor = new $class();
+		}
+
+		return self::$monitor;
+	}
+
+	/**
+	 * Initialize.
+	 *
+	 * @return void
+	 */
+	public function init() {
+		$this->threshold_seconds       = absint( apply_filters( 'action_scheduler_pastdue_actions_seconds', DAY_IN_SECONDS ) );
+		$this->threshold_minimum       = absint( apply_filters( 'action_scheduler_pastdue_actions_min', 1 ) );
+		$this->interval_check          = absint( apply_filters( 'action_scheduler_pastdue_actions_check_interval', ( $this->threshold_seconds / 4 ) ) );
+		$this->interval_email_seconds  = absint( apply_filters( 'action_scheduler_pastdue_actions_email_interval', HOUR_IN_SECONDS ) );
+		$this->threshold_email_minimum = absint( apply_filters( 'action_scheduler_pastdue_actions_email_min', $this->threshold_minimum ) );
+		$this->notify_methods          = $this->notify_methods();
+
+		$notify_email_to = apply_filters( 'action_scheduler_pastdue_monitor_notify_email_to', get_site_option( 'admin_email' ) );
+
+		if ( ! is_string( $notify_email_to ) || ! is_email( $notify_email_to ) ) {
+			trigger_error( 'Invalid email address provided for past-due actions flooded email notification: using site\'s administrator email address.', E_USER_WARNING );
+			$notify_email_to = get_site_option( 'admin_email' );
+		}
+
+		$this->notify_email_to = $notify_email_to;
+
+		add_action( 'action_scheduler_stored_action', array( $this, 'on_action_stored' ) );
+
+		if ( is_admin() && ( ! defined( 'DOING_AJAX' ) || empty( DOING_AJAX ) ) ) {
+			add_action( 'admin_notices', array( $this, 'action__admin_notices' ) );
+		}
+	}
+
+	/**
+	 * Check if notification method enabled.
+	 *
+	 * @param string $method
+	 * @return bool
+	 */
+	protected function notify( $method ) {
```

Comment:
This method name is a bit misleading, it seems to suggest that a notification will be sent. What about something like `notify_methods_has` instead?